methods/guest/src/bin/is_even.rs

```diff
@@ -22,12 +22,14 @@ risc0_zkvm::guest::entry!(main);
 
 pub fn main() {
     // Read data sent from the application contract.
-    let mut number: U256 = env::read();
+    let number: U256 = env::read();
 
     // Run the computation.
-    if number.bit(0) {
-        number = U256::ZERO;
-    }
+    // In this case, asserting that the provided number is even.
+    assert!(
+        number != U256::ZERO && number.bit(0) == false,
```

Comment:
Zero is actually a valid input. It might be useful to e.g. reset the stored number to 0 in the case that it was set to different number. ("useful" is a bit generous for this example, but you get the point, haha)
```suggestion
        number.bit(0) == false,
```